modules/programs/vscode/haskell.nix

```diff
@@ -0,0 +1,102 @@
+{ pkgs, config, lib, ... }:
+let
+
+  defaultHieNixSourceParams = {
+    owner = "domenkozar";
+    repo = "hie-nix";
+    rev = "6794005f909600679d0b7894d0e7140985920775";
+    sha256 = "0pc90ns0xcsa6b630d8kkq5zg8yzszbgd7qmnylkqpa0l58zvnpn";
+  };
+  defaultHieNixSource = pkgs.fetchFromGitHub defaultHieNixSourceParams;
+
+  defaultHieNixExe = hie-nix.hies + "/bin/hie-wrapper";
+  defaultHieNixExeText = ''hie-nix.hies + "/bin/hie-wrapper"'';
+
+  inherit (pkgs.vscode-utils) buildVscodeMarketplaceExtension;
+  inherit (lib) types;
+  cfg = config.programs.vscode.haskell;
+
+  hieNixArgs = lib.optionalAttrs (!cfg.hieNixUsePinnedPkgs) {
+    inherit pkgs;
+  };
+  hie-nix = import cfg.hieNixSource hieNixArgs;
+
+in
+{
+  options.programs.vscode.haskell.enable = lib.mkEnableOption "Haskell integration for Visual Studio Code";
+
+  options.programs.vscode.haskell.hieNixSource = lib.mkOption {
+    type = lib.types.path;
+    default = defaultHieNixSource;
+    defaultText = lib.literalExample ''
+      pkgs.fetchFromGitHub
+        ${lib.generators.toPretty {} defaultHieNixSourceParams}
+    '';
+    example = lib.literalExample ''
+      ~/src/haskell-ide-engine
+    '';
+    description = ''
+      A version of the <link xlink:href="https://github.com/domenkozar/hie-nix/">hie-nix sources</link>.
+    '';
+  };
+
+  options.programs.vscode.haskell.hieExecutablePath = lib.mkOption {
```

Comment:
Seems to me like this is the option that matters. Perhaps one could simply remove the `hieNixSource` and `hieNixUsePinnedPkgs` options? Setting up the package seems like something the user should do in an overlay or `packageOverrides`.